### ISSUE TITLE
Highlighting peculiarities

### ISSUE THREAD

**Issue by Dev-iL** [Gimly/vscode-matlab#7]
See examples in the image below:

![capture](https://cloud.githubusercontent.com/assets/6509619/17475218/829b1d54-5d62-11e6-9f23-b27d8ec315bd.PNG)

Some ideas on how to solve this can be taken from [this repository](https://github.com/amroamroamro/prettify-matlab/).

**Comment by Dev-iL** [comment]
False alarm; my bad.

**Comment by Gimly** [comment]
Thanks for the bug report, I'll try to figure out a way to fix this.

**Comment by Gimly** [comment]
Those are some really weird issues. I'm using the same syntax as Textmate, do you know if this is working in this editor?

Multiline comments are quite tricky to get right as I can see, Matlab's documentation says:
> The %{ and %} operators must appear alone on the lines that immediately precede and follow the block of help text. Do not include any other text on these lines.

I'll have to figure this out. Matlab's text editor treats the multiline comment as a line comment if there is something else on the same line or if there is no closing of this multiline comment.

I'm not really sure that VS Code's syntax will be able to take this into account. Interesting find.

**Comment by Gimly** [comment]
The issue with the switch-case has been fixed by the fixing of issue #9.

![image](https://cloud.githubusercontent.com/assets/168669/23400778/693bf23c-fda5-11e6-9910-bbdd3753f482.png)

As you can see there is still the issue with the block comments. I'm not quite sure what the correct syntax highlighting is in that case.

**Comment by sco1** [comment]
MATLAB appears to just treat comment blocks that do not follow the syntax you quoted as regular comments

For example: 
```
function output = highlights(input)
% This is a single line comment

%{
block
comment
%}

%{} A single line comment
not a comment

%{ Also a single line comment
not a comment
%} Despite this closure

%{ Should be treated as a single line comment }
function1_call();

function2_call(); %{ Should also be treated as a single line comment }
function3_call();
```

MATLAB's handling:

![MATLAB](https://cloud.githubusercontent.com/assets/5323929/23404964/c6ac36b8-fd85-11e6-80b5-8a72d8b13178.PNG)

VSCode (MATLAB 0.5):

![VSCode](https://cloud.githubusercontent.com/assets/5323929/23404991/eba529a2-fd85-11e6-8617-ac4de84ef83b.PNG)

FWIW Sublime's default syntax highlighter and StackOverflow's both also handle this incorrectly. The latter is presumably based on Amro's but I don't have the immediate knowledge to test.

**Comment by Gimly** [comment]
@sco1 Thanks for the test ! So according to the way matlab is handling the block comments, a block comment operator is only valid if it's the only thing on the line. If it's not the case, it should be treated as a single line comment.

I should probably be able to implement this correctly now that I have the example from Matlab.
Thanks

**Comment by Dev-iL** [comment]
@Gimly if only I had known _that_ was what you were missing all along :)

**Comment by Gimly** [comment]
@Dev-iL Don't worry, the real ressource I was/am missing was time...

**Comment by subhammridha** [comment]
Hello, 

I don't know much about regex, but I think I have figured out a way to rectify this problem.

In line matlab.tmLanguage file [https://github.com/Gimly/vscode-matlab/blob/master/syntaxes/matlab.tmLanguage#L1404](url), if we replace the existing syntax for finding block code (,i.e., `%\{`) by `(^[ \t]+)?(?=%\{\n)`, syntax highlighting of block code seems to be working fine. I had modified this using the regex for finding line comments. As per my testing this seems to be working fine.

Further scrutiny and implementation is needed at your end.

Kindly check it.

**Comment by Gimly** [comment]
This issue should be fixed with version 0.6.0. Please reopen if there is another issue.
